front/src/common/Map/Search/MapSearchStation.jsx

```diff
@@ -1,72 +1,89 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { useSelector, useDispatch } from 'react-redux';
 import { updateMapSearchMarker } from 'reducers/map';
 import { useTranslation } from 'react-i18next';
 import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
-import { post } from 'common/requests';
 import { useDebounce } from 'utils/helpers';
+import nextId from 'react-id-generator';
 import turfCenter from '@turf/center';
 import StationCard from 'common/StationCard';
 import { getInfraID } from 'reducers/osrdconf/selectors';
-import nextId from 'react-id-generator';
-import { SEARCH_URL } from '../const';
+import { getMap } from 'reducers/map/selectors';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 
 export default function MapSearchStation(props) {
   const { updateExtViewport } = props;
-  const map = useSelector((state) => state.map);
-
+  const map = useSelector(getMap);
   const [searchState, setSearch] = useState('');
-  const [dontSearch, setDontSearch] = useState(false);
   const [searchResults, setSearchResults] = useState(undefined);
+  const [trigramResults, setTrigramResults] = useState([]);
+  const [nameResults, setNameResults] = useState([]);
   const infraID = useSelector(getInfraID);
 
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
   const dispatch = useDispatch();
 
   const { t } = useTranslation(['map-search']);
 
-  const updateSearch = async (params) => {
-    try {
-      const data = await post(SEARCH_URL, params);
-      setSearchResults(data);
-    } catch (e) {
-      /* empty */
-    }
-  };
-
   const debouncedSearchTerm = useDebounce(searchState, 300);
 
-  const getPayload = () => {
-    let trigram = null;
-    let name = null;
-    if (!dontSearch && debouncedSearchTerm) {
-      if (searchState.length < 3) {
-        trigram = searchState;
-        name = null;
-      } else if (searchState.length === 3) {
-        trigram = searchState;
-        name = searchState;
-      } else if (searchState.length > 3) {
-        name = searchState;
-        trigram = null;
-      }
+  // Create playload based on the type of search "name" or "trigram"
+
+  const createPayload = (searchQuery) => ({
+    object: 'operationalpoint',
+    query: ['and', searchQuery, ['=', ['infra_id'], infraID]],
+  });
+
+  // Sort on name, and on yardname
+  const orderResults = (results) =>
+    results.sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
+
+  const searchByTrigrams = useCallback(async () => {
+    const searchQuery = ['=i', ['trigram'], searchState];
+    const payload = createPayload(searchQuery);
+    const { data, error } = await postSearch({
+      body: payload,
+    });
+    if (error) console.error(error);
+    setTrigramResults([...data]);
```

Comment:
```suggestion
    if (error) {
      resetSearchResult()
    } else {
      setTrigramResults(data);
    }
```

with :
```
  const resetSearchResult = () => {
    setNameResults([]);
    setTrigramResults([]);
  };
```
The SearchResults will be reset in the useEffect.

Also, no need to console.log the error as the network will print it in the console.

Console.logs when the api call fails:
![image](https://user-images.githubusercontent.com/45000526/226630161-98256118-3a7e-44d6-a870-757022c34bcd.png)